crates/control/src/behavior/support.rs

```diff
@@ -83,13 +83,22 @@ fn support_pose(
     let filtered_game_state = world_state
         .filtered_game_controller_state
         .map(|filtered_game_controller_state| filtered_game_controller_state.game_state);
-    let mut clamped_x = match filtered_game_state {
-        Some(FilteredGameState::Ready { .. })
-        | Some(FilteredGameState::Playing {
-            ball_is_free: false,
-            kick_off: true,
-            ..
-        }) => supporting_position
+    let sub_state = world_state
+        .filtered_game_controller_state
+        .map(|filtered_game_controller_state| filtered_game_controller_state.sub_state);
+    let mut clamped_x = match (filtered_game_state, sub_state) {
+        (Some(FilteredGameState::Ready { .. }), Some(Some(SubState::PenaltyKick))) => {
```

Comment:
What is the difference between filtered_game_state and world_state.filtered_game_controller_state?

Reply:
`filtered_game_state` is the `world_state.filtered_game_controller_state.game_state` which is first properly handled in line 83